### PR TITLE
Feat phpstan lvl2

### PR DESCRIPTION
**Pull request by Rom1-B**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -

Probably the last round for 10.0/bf, because for the remaining cases, I don't see how to treat them without risking BC Break.

### CODE REVIEW

File: phpstan-baseline.neon
Comment (trasher) [inline phpstan-baseline.neon:1]: Why this new file?

Comment (Rom1-B) [reply phpstan-baseline.neon:1]: This is the baseline file that allows me to list the remaining errors to monitor progress. I've just deleted it.

File: src/Database.php [hunk 1]
```diff
@@ -374,6 +374,14 @@ public static function rawSearchOptionsToAdd()
         return $tab;
     }
 
+    /**
```
Comment (trasher) [inline src/Database.php:377]: I'm not fans of duplicated phpdoc when not necessary.

I guess warning will be solved if you use `get_class($item)` interad of `getType()` (probably same for all other ocurrences)

**Review by AdrienClairembault — APPROVED** [review]

**Review by trasher — CHANGES_REQUESTED** [review]
See comment

File: src/Database.php (hunk 1 above)
Comment (Rom1-B) [reply src/Database.php]: This is the only case where using get_class() is sufficient.

File: src/ITILFollowup.php [hunk 2]
```diff
@@ -890,6 +890,13 @@ public static function showMassiveActionsSubForm(MassiveAction $ma)
         return parent::showMassiveActionsSubForm($ma);
     }
 
+    /**
+     * @see CommonDBTM::processMassiveActionsForOneItemtype()
```
Comment (cconard96) [inline src/ITILFollowup.php:894]: Check item type inside method and add KO status for all IDs if it isn't CommonITILObject.

File: src/DatabaseInstance.php [hunk 3]
```diff
@@ -537,6 +537,14 @@ public function pre_purgeInventory()
         return true;
     }
 
+    /**
+     * @see CommonGLPI::getTabNameForItem()
```
[the remark this answers is not in the document]
Comment (trasher) [reply src/DatabaseInstance.php:541]: Seems not enough from my test. Another solution is to check `$item instanceof CommonDBTM`

**Review by trasher — APPROVED** [review]

**Review by cconard96 — APPROVED** [review]

**Review by orthagh — APPROVED** [review]

**Pull request merged**